packages/core/src/jsonld/documentLoader.ts

```diff
@@ -8,35 +8,46 @@ type CachedDocuments = {
   [index: string]: any;
 };
 
-export const createDocumentLoader = (
-  registry: IdenityRegistry,
-  {
-    documents = {},
-  }: {
-    documents?: CachedDocuments;
-  } = {},
-) => {
-  return async (url: string) => {
-    const context = documents[url];
-    if (context !== undefined) {
-      return {
-        contextUrl: null,
-        documentUrl: url,
-        document: context,
-      };
-    }
-
-    if (url.startsWith('did:tangleid:')) {
-      const document = await registry.fetch(url);
-      return {
-        document,
-        documentUrl: url,
-        contextUrl: null,
-      };
-    }
-
-    const result = await nodeDocumentLoader(url);
-    documents[url] = result.document;
-    return result;
-  };
-};
+export class DocumentLoader {
+  registry: IdenityRegistry;
+  documents: CachedDocuments = {};
+
+  constructor(registry: IdenityRegistry) {
+    this.registry = registry;
+  }
+
+  getDocuments() {
+    return this.documents;
+  }
+
+  setDocuments(documents: CachedDocuments) {
+    this.documents = { ...this.documents, ...documents };
+  }
+
+  loader() {
+    return async (url: string) => {
+      const context = this.documents[url];
+      if (context !== undefined) {
+        return {
+          contextUrl: null,
+          documentUrl: url,
+          document: context,
+        };
+      }
+
+      if (url.startsWith('did:tangleid:')) {
```

Comment:
Should it be `did:tangle`?

Reply:
We will use the `did:tangle` when we implement the DID method with the MAM v2.
I'll suggest keeping the `did:tangleid` for now.

Reply:
No, let's migrate now. We don't have to ensure the compatibility at this moment. Minimizing technical support efforts is crucial.

Reply:
I'll send another PR to change the DID method name. 👌

Reply:
Fixed. 👍